### DISCUSSION TITLE
How to convert all images in directories and subdirectories into webp format

### DISCUSSION THREAD

**Discussion by nikepoter** [ImageMagick/ImageMagick#7711] [Help]
### ImageMagick version

7.1.1

### Operating system, version and so on

Windows

### Description

How to convert all images in directories and subdirectories into webp format

**Comment by dlemstra** [comment]
What have you tried yourself and at what point are you stuck?

**Reply by nikepoter** [reply]
`for /f "delims=" %i in ('dir /b /a-d') do (magick mogrify -quality 80 -format webp *.jpg）`

This code can only process the current directory, subdirectories are not executed

**Comment by snibgo** [comment]
```
dir /b /a-d
```
This lists all the files in the current directory. If you want to also include file in subdirectories, add "/s".

```
magick mogrify -quality 80 -format webp *.jpg
```
Each time this is executed, this processes all `*.jpg` files in the current directory.

The `magick mogrify` command is executed once for every output from the `dir`. So the overall command processes every jpeg file multiple times, once for each output from the `dir`.

Instead of processing `*.jpg`, it would seem more sensible to process `%i`.
